frontend/src/pages/question/questionBuilder/QuestionBuilder.jsx

```diff
@@ -77,53 +78,45 @@ export default function QuestionBuilder(props) {
    * @param {Number} question.max_connectivity max connections for question
    */
   function onQuestionTemplate(question) {
-    Object.values(question.query_graph.edges).forEach(queryGraphUtils.standardizeType);
-    Object.values(question.query_graph.nodes).forEach(queryGraphUtils.standardizeType);
+    Object.values(question.query_graph.edges).forEach(queryGraphUtils.standardizePredicate);
+    Object.values(question.query_graph.nodes).forEach(queryGraphUtils.standardizeCategory);
+    Object.values(question.query_graph.nodes).forEach(queryGraphUtils.standardizeIDs);
     questionStore.updateQueryGraph(question.query_graph);
     questionStore.updateQuestionName(question.natural_question);
     setStep('build');
     toggleModal(false);
   }
 
   function validateAndParseFile(fileContents) {
-    let fileContentObj;
+    let message;
     try {
-      fileContentObj = JSON.parse(fileContents);
+      message = JSON.parse(fileContents);
     } catch (err) {
       displayAlert('error',
         'The provided file is not valid JSON. Please fix and try again.');
       setStep('options');
       return;
     }
 
-    if (!('query_graph' in fileContentObj)) {
-      displayAlert('error',
-        'The provided file does not have a query_graph object. Please ensure that the file format adheres to the Reasoner Standard API Query Schema.');
-      setStep('options');
-      return;
-    }
+    const validationErrors = trapiUtils.validateMessage(message);
 
-    if (!_.isArray(fileContentObj.query_graph.nodes) ||
-       !_.isArray(fileContentObj.query_graph.edges)) {
-      displayAlert('error',
-        'The provided file does not have an array of nodes or edges. Please ensure that the file format adheres to the Reasoner Standard API Query Schema.');
+    if (validationErrors.length) {
+      displayAlert('error', `Found errors while parsing message: ${validationErrors.join(', ')}`);
       setStep('options');
       return;
     }
 
-    // TODO: this is pretty hacky
-    fileContentObj.query_graph.nodes.forEach((node) => {
-      if (!node.label) {
-        node.label = `${node.id}: ${node.name || node.curie || node.type}`;
+    const { query_graph } = message.message;
+
+    Object.values(query_graph.nodes).forEach((node) => {
+      if (!node.name) {
+        node.name = `${node.id || strings.displayCategory(node.category)}`;
       }
-      queryGraphUtils.standardizeType(node);
+      queryGraphUtils.standardizeCategory(node);
     });
+    Object.values(query_graph.edges).forEach((e) => queryGraphUtils.standardizePredicate(e));
 
-    fileContentObj.query_graph.edges.forEach(queryGraphUtils.standardizeType);
-
-    questionStore.updateQueryGraph(
-      queryGraphUtils.convert.reasonerToInternal(fileContentObj.query_graph),
-    );
+    questionStore.updateQueryGraph(query_graph);
```

Comment:
I agree this is annoying and I'd like to fix it. But it gets messy because the graph wasn't updating when it should have, so now it rerenders too much. In the new UI design, we're going to be redoing the whole graph visualization, so I think we can push this off.